include/openenclave/internal/plugin.h

```diff
@@ -0,0 +1,465 @@
+
+    /**
+     *
+     * Gets the optional settings data for thhe input evidence format.
+     *
+     * @experimental
+     *
+     * @param[in] context A pointer to the verifier plugin struct.
+     * @param[in] format The format for which to retrieve the optional settings.
+     * @param[out] settings An output pointer that will be assigned the address
+     * of a dynamically allocated buffer that holds the returned settings. This
+     * pointer can be assigned a NULL value if there is no settings needed.
+     * @param[out] settings_size A pointer that points to the size of the
+     * returned format settings buffer (number of bytes).
+     * @retval OE_OK on success.
+     * @retval OE_INVALID_PARAMTER At least one of the parameters is invalid.
+     * @retval other appropriate error code.
+     */
+    oe_result_t (*get_format_settings)(
+        oe_verifier_t* context,
+        uint8_t** settings,
+        size_t* settings_size);
+
+    /**
+     * Verifies the attestation evidence and returns the claims contained in
+     * the evidence.
+     *
+     * Each plugin must return the following required claims:
+     *  - id_version (uint32_t)
+     *      - Version number. Must be 1.
+     *  - security_version (uint32_t)
+     *      - Security version of the enclave. (ISVN for SGX).
+     * - attributes (uint64_t)
+     *      - Attributes flags for the evidence:
+     *          - OE_REPORT_ATTRIBUTES_DEBUG: The evidence is for a debug
+     * enclave.
+     *          - OE_REPORT_ATTRIBUTES_REMOTE: The evidence can be used for
+     * remote attestation.
+     * - unique_id (uint8_t[32])
+     *      - The unique ID for the enclave (MRENCLAVE for SGX).
+     * - signer_id (uint8_t[32])
+     *      - The signer ID for the enclave (MRSIGNER for SGX).
+     * - product_id (uint8_t[32])
+     *      - The product ID for the enclave (ISVPRODID for SGX).
+     * - validity_from (oe_datetime_t)
+     *      - Overall datetime from which the evidence and endorsements are
+     *        valid.
+     * - validity_until (oe_datetime_t)
+     *      - Overall datetime at which the evidence and endorsements expire.
+     * - format_uuid (uint8_t[16])
+     *      - The format UUID of the verified evidence.
+     *
+     * The plugin is responsible for handling endianness and ensuring that the
+     * data from the raw evidence converted properly for each platform.
+     *
+     * @experimental
+     *
+     * @param[in] context A pointer to the verifier plugin struct.
+     * @param[in] evidence_buffer The evidence buffer.
+     * @param[in] evidence_buffer_size The size of evidence_buffer in bytes.
+     * @param[in] endorsements_buffer The endorsements buffer.
+     * @param[in] endorsements_buffer_size The size of endorsements_buffer in
+     * bytes.
+     * @param[in] policies A list of policies to use.
+     * @param[in] policies_size The size of the policy list.
+     * @param[out] claims The list of base + custom claims.
```

Comment:
Are these claims that appear in the evidence (and/or endorsements)?
Or are these claims that should be used as attestation results (which can be different)?